### PR TITLE
Handles rerouting of non-standard/feature articles in amp

### PR DESCRIPTION
**Pull request by ashleyjelks**
This PR fixes a bug causing Artsy Vanguard AMP links to return a 404, by redirecting all non-standard or non-feature AMP links back to `/article-original-slug.`

[Links to GROW-1546](https://artsyproduct.atlassian.net/browse/GROW-1546)

### CODE REVIEW

**Comment by joeyAghion** [comment]
Did you already consider not rendering the `<link rel="amphtml"...` meta at all for articles that don't support it? We use an `article.hasAMP()` helper for this elsewhere that either has a bug or isn't being used correctly in this case:

https://github.com/artsy/force/blob/002ce94a5e59b3d3a28d5ae1adcbe6c795d9349d/src/desktop/apps/article/templates/classic_meta.jade#L3-L4

**Comment by zephraph** [comment]
This step is okay because it solves any in the wild 404s, but @joeyAghion is right here. I'm somewhat worried that if we advertise that we have an amp page but the  page is reported as invalid we'll get dinged by google for that.

File: src/desktop/apps/article/templates/classic_meta.jade [hunk 1]
```diff
@@ -1,6 +1,6 @@
 title= article.get('search_title') ? article.get('search_title') : article.get('thumbnail_title')
 link( rel="canonical" href=article.fullHref() )
-if article.hasAMP()
+if article.hasAMP() && (article.layout == "standard" || article.layout == "feature" || article.layout == "news")
```
Comment (zephraph) [inline src/desktop/apps/article/templates/classic_meta.jade]: Is there a reason not to add this logic directly to `hasAMP()`?

https://github.com/artsy/force/blob/2bc3466f25328fb2a9956cb01a71e1e1f0799431/src/desktop/models/article.coffee#L222-L228

File: src/desktop/apps/article/templates/meta.jade [hunk 2]
```diff
@@ -14,7 +15,7 @@
 title= title + titleExtension
 meta( name='description', content= search_description )
 link( rel='canonical' href=url )
-if article.featured && article.published
+if article.featured && article.published  && isAMPEligible
```
Comment (zephraph) [inline src/desktop/apps/article/templates/meta.jade]: Would it make sense for this to be `article.hasAMP()`?

File: src/desktop/apps/article/templates/classic_meta.jade (hunk 1 above)
Comment (ashleyjelks) [reply src/desktop/apps/article/templates/classic_meta.jade]: `hasAMP`  is already checking for article layout being published, featured, and of layout type news || standard || feature, yet it still returns true for `hasAMP` for Vanguard articles, but it isn't clear to me why.  The logic here seems to be correct though, which is why I didn't modify this part.

Comment (ashleyjelks) [reply src/desktop/apps/article/templates/classic_meta.jade]: The data for Vanguard articles is `featured: true`, `layout: "series"`, `published: true`.

Comment (eessex) [reply src/desktop/apps/article/templates/classic_meta.jade]: Articles in the `classic` layout are only from internal/partner channels, so should already be excluded from AMP-- I don't think we will ever reach this template, the `hasAMP` may be legacy from before we moved articles to reaction.

File: src/desktop/apps/article/templates/meta.jade (hunk 2 above)
Comment (eessex) [reply src/desktop/apps/article/templates/meta.jade]: Agreed, we should use the model's `hasAMP` here.

File: src/desktop/apps/article/templates/meta.jade [hunk 3]
```diff
@@ -14,7 +14,7 @@
 title= title + titleExtension
 meta( name='description', content= search_description )
 link( rel='canonical' href=url )
-if article.featured && article.published
+if article.featured && article.published
```
Comment (eessex) [inline src/desktop/apps/article/templates/meta.jade]: Looks like there is maybe an extra empty space at the end of this line

File: src/desktop/test/models/article.test.coffee [hunk 4]
```diff
@@ -326,6 +326,19 @@ describe "Article", ->
         layout: 'series'
       @article.hasAMP().should.be.false()
 
+    it 'returns false if article is not featured', ->
+      @article.set
+        sections: [ type: 'text' ]
+        featured: false
+      @article.hasAMP().should.be.false()
+
+    it 'returns false if article is a series', ->
```
Comment (eessex) [inline src/desktop/test/models/article.test.coffee]: We should add a test here for `video` layout as well to make sure all our bases are covered.

**Pull request merged**